src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,270 @@
+    String op = basicOp(termNode);
+    return field.getColumn() + op + pgTerm;
+  }
+
+  String handleTypeText(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String base = termNode.getRelation().getBase();
+    boolean ft = "=".equals(base) || "all".equals(base);
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      if (c == '\'' && !backslash) {
+        pgTerm.append('\''); // important to avoid SQL injection
+      } else if (c == '*' && ft) {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op * unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '?' && ft) {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op ? unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '^' && ft) {
+        if (!backslash) {
+          throw new IllegalArgumentException("Anchor op ^ unsupported for: " + termNode.toCQL());
+        }
+      } else if (backslash) {
+        pgTerm.append('\\'); // pass-tru the backslash for Postgres to honor (including \')
+      }
+      if (c == '\\') {
+        backslash = true;
+      } else {
+        backslash = false;
+        pgTerm.append(c);
+      }
+    }
+    if (backslash) {
+      pgTerm.append('\\');
+    }
+    if (!ft) {
+      return field.getColumn() + " " + basicOp(termNode) + " E'" + pgTerm + "'";
+    }
+    return "to_tsvector('" + language + "', " + field.getColumn() + ") @@ plainto_tsquery('"
+        + language + "', E'" + pgTerm + "')";
+  }
+
+  String handleTypeNumber(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String cqlTerm = termNode.getTerm();
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      switch (c) {
+        case '.':
+        case ',':
+        case '+':
+        case '-':
+          break;
+        default:
+          if (!Character.isDigit(c)) {
+            throw new IllegalArgumentException("Bad numeric for: " + termNode.toCQL());
+          }
+      }
+    }
+    return field.getColumn() + numberOp(termNode) + cqlTerm;
+  }
+
+  String handleTypeBoolean(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String cqlTerm = termNode.getTerm();
+    String pgTerm;
+    if ("false".equalsIgnoreCase(cqlTerm)) {
+      pgTerm = "FALSE";
+    } else if ("true".equalsIgnoreCase(cqlTerm)) {
+      pgTerm = "TRUE";
+    } else {
+      throw new IllegalArgumentException("Bad boolean for: " + termNode.toCQL());
+    }
+    return field.getColumn() + basicOp(termNode) + pgTerm;
+  }
+
+  String whereHandle(CQLNode node) {
```

Comment:
`whereHandle` is an odd name for this function: all the others _start_ with `where`, so why not `handleWhere` or `handleNode` or similar?

Reply:
Renamed.